tests/socket-select.phpt

```diff
@@ -0,0 +1,40 @@
+--TEST--
+Test parameter handling in Socket::select().
+--FILE--
+<?php
+    $socket = new Socket(\Socket::AF_INET, \Socket::SOCK_STREAM, \Socket::SOL_TCP);
+    $socket->listen();
+
+    try {
+        Socket::select();
+    } catch(Throwable $throwable) {
+        var_dump($throwable->getMessage());
+    }
+    $read   = [$socket];
+    $write  = null;
+    $except = null;
+
+    // Valid arguments, return immediately
+    var_dump(Socket::select($read, $write, $except, 0));
+
+    $read = [$socket];
+
+    // Valid sec argument, wait 1 second
+    var_dump(Socket::select($read, $write, $except, 1, 0, $errno));
+    var_dump($errno);
+
+    $read = [$socket];
+
+    // Invalid sec argument, return immediately
+    var_dump(Socket::select($read, $write, $except, -1, 0, $errno));
+    var_dump($errno);
+
+    $socket->close();
+--EXPECTF--
+
+Warning: Socket::select() expects at least 4 parameters, 0 given in %s on line %d
+int(0)
+int(0)
+int(0)
+bool(false)
+int(22)
```

Comment:
This test does not work as expected on Windows (I'm getting all zeros, no errors) - not sure if this is a Windows-specific bug or what, haven't investigated in-depth yet.

Reply:
On win32, a negative timeout seems to be handled like a zero timeout, without any error.